src/app/(tabs)/profile.tsx

```diff
@@ -5,14 +5,36 @@ import Center from "@src/components/ui/Center";
 import Stack from "@src/components/ui/Stack";
 import { useUser } from "@src/state/useUser";
 import { Redirect } from "expo-router";
+import Button from "@src/components/ui/Button";
+import { useRouter } from "expo-router";
 
 const ProfileScreen = () => {
   const isAuth = useUser((data) => data.isAuth);
+  const router = useRouter();
 
   if (!isAuth) {
     return <Redirect href="/provider-signup" />;
   }
 
+  const logout = () => {
+    const user = {
```

Comment:
no need to pass the user, just make the user params optional in the setAuth function